### PR TITLE
[#9258] [Platform] Some fields of Edit Cloud Provider Configuration are NOT greyed out 

### PR DESCRIPTION
**Pull request by rajmaddy89**
**ISSUE 9258**
Assuming a cloud provider configuration exists, go to 'Yugabyte DB Admin Console' -> Configs -> 'Edit Configuration'.
Ideally, when the edit page opens, all text fields except 'Hosted Zone ID' should be greyed out. 

This PR fixes 2 things:
1. 'Name' and 'SSH Key' are greyed
2. Submit button will throw toast error message

### CODE REVIEW

**Review by yb-andrew — CHANGES_REQUESTED** [review]
Should try to figure out why the `providersInUse` variable is not including the provider. Also, please include the github issue number in the title: `[#9258][Platform] Etc etc.`

File: managed/ui/src/components/universes/UniverseConnectModal/UniverseConnectModal.js [hunk 1]
```diff
@@ -154,6 +161,13 @@ class UniverseConnectModal extends Component {
         </Fragment>
       );
       const connectIp = this.state.connectIp || '127.0.0.1';
+      const jdbcConnection = `jdbc:postgresql://${connectIp}:${ysqlRpcPort}/yugabyte`;
+      const jdbcTLSConnection = jdbcConnection + sslModeRequired;
+      const ysqlConnection = 'bin/ysqlsh';
+      const ySqlTLSConnection = ysqlConnection + sslModeRequired;
```
[the remark this answers is not in the document]
Comment (rajmaddy89) [reply managed/ui/src/components/universes/UniverseConnectModal/UniverseConnectModal.js:167]: According to @yb-andrew 's comment: 
```
The connect dialog should check the universe details for the TLS enabled setting. Should be two settings called enableNodeToNodeEncrypt and enableClientToNodeEncrypt, if either is true then we have enabled TLS encryption for the universe.
```
Refer to: https://github.com/yugabyte/yugabyte-db/pull/9473#pullrequestreview-720036413

File: managed/ui/src/components/config/PublicCloud/views/EditProvider/EditProviderForm.js [hunk 2]
```diff
@@ -83,7 +84,7 @@ export default class EditProviderForm extends Component {
               type="text"
               label="Name"
               component={YBTextInputWithLabel}
-              isReadOnly={providersInUse.has(uuid)}
+              isReadOnly={true}
```
[the remark this answers is not in the document]
Comment (rajmaddy89) [reply managed/ui/src/components/config/PublicCloud/views/EditProvider/EditProviderForm.js:87]: @yb-andrew : Should this `providersInUse` be Universe UUID or AWS Provider configuration UUID

The problem currently thats happening is `providersInUse.has(uuid)`, the `providersInUse` has AWS Provider configuration UUID and the parameter to the function `has` is `uuid` which here refers to Universe UUID, this is the confusiin

[the remark this answers is not in the document]
Comment (rajmaddy89) [reply managed/ui/src/components/config/PublicCloud/views/EditProvider/EditProviderForm.js:87]: image

The problem is the UUID at index 1 and index 2 are compared, but dont have any provider with name `cloud-1` as shown. Not sure why this mysterious provider at index 1 is showing up

[the remark this answers is not in the document]
Comment (rajmaddy89) [reply managed/ui/src/components/config/PublicCloud/views/EditProvider/EditProviderForm.js:87]: Also in this PR: https://github.com/yugabyte/yugabyte-db/commit/90e6a9f7f425d935861926fe9c9d1253b67cab24

I see that you are saying the followinf steps:
`
1. Create new provider.
2. try editing it immediately,.
3 As no universe is using it you should be able to change config.
4. Try similar steps with provider in use and go to edit provider config.
5. User should not be able to to change provider config.
`

In my case I created a AWS provider config but the Universe is not using , may be thats why I am able to edit it ?? What do you think about that

[the remark this answers is not in the document]
Comment (yb-andrew) [reply managed/ui/src/components/config/PublicCloud/views/EditProvider/EditProviderForm.js:87]: @rajmaddy89 The provider called `cloud-1` is related to the yugabyte daemon (`yugabyted`) program that is separate from the platform for the most part. The function `getClusterProviderUUIDs` is used to find the provider UUID and is given a list of universes, so this list of provider UUID's is for providers that are currently in use. So far, this is working as intended. What needs to be fixed in this case is allowing the user to submit a change to the provider when the (AWS) provider  is not in use.

File: managed/ui/src/components/universes/UniverseConnectModal/UniverseConnectModal.js [hunk 3]
```diff
@@ -124,6 +124,13 @@ class UniverseConnectModal extends Component {
       const {
         universeDetails: { clusters, communicationPorts }
       } = universeInfo;
+
+      const tlsAvailability = getFeatureState(
+        currentCustomer.data.features,
+        'universes.details.overview.manageEncryption'
+      );
+      const isTLSEnabled = isEnabled(tlsAvailability);
```
Comment (yb-andrew) [inline managed/ui/src/components/universes/UniverseConnectModal/UniverseConnectModal.js:132]: The way to check if TLS is enabled for a universe is the metadata of the universe details.
Redux state:
From `state.universe.currentUniverse.data.universeDetails.clusters`, get the primary cluster's `userIntent` and check whether `enableNodeToNodeEncrypt` or `enableClientToNodeEncrypt` flags are set to true. If either flag (or both) are set to true, TLS is enabled for the universe.

**Review by yb-andrew — CHANGES_REQUESTED** [review]
Your way of checking whether TLS is enabled is incorrect. The information is present on the universe details, in the userIntent. See comment.

File: managed/ui/src/components/universes/UniverseConnectModal/UniverseConnectModal.js (hunk 3 above)
Comment (rajmaddy89) [reply managed/ui/src/components/universes/UniverseConnectModal/UniverseConnectModal.js:132]: @yb-andrew : I have already fixed it the right way in https://github.com/yugabyte/yugabyte-db/pull/9473, so when this PR: 9441 is merged first and then when [9473](https://github.com/yugabyte/yugabyte-db/pull/9473/files) is merged, we should not have any confusion. Please let me know. Thanks.

**Review by yb-andrew — APPROVED** [review]
As per our conversation, this PR is to be merged first and then the next PR will contain a change to fix the incorrect TLS enabled access.

**Pull request merged**